### PR TITLE
[RFR] Logfile fetch and decode

### PR DESCRIPTION
**Pull request by SimonBiggs**
Automatically fetch and index log files directly from the Linac backup shared drive.

### CODE REVIEW

**Comment by SimonBiggs** [comment]
This isn't failing locally. I think I might investigate this by swapping to the alternative pipeline method using tox first.

**Comment by Centrus007** [comment]
Is this still RFR? Or are you still diagnosing those two test failures?

**Comment by SimonBiggs** [comment]
Yup, I believe that should be fixed by https://github.com/pymedphys/pymedphys/pull/110, see https://github.com/kynan/nbstripout/issues/94

File: src/pymedphys/cli/dicom.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+# You should have received a copy of the Apache-2.0 along with this
+# program. If not, see <http://www.apache.org/licenses/LICENSE-2.0>.
+
+
+from ..dicom import (
+    adjust_machine_name_cli, adjust_RED_cli,
+    adjust_RED_by_structure_name_cli)
+
+
+def dicom_cli(subparsers):
```
Comment (Centrus007) [inline src/pymedphys/cli/dicom.py]: docs showing intended use?

Comment (Centrus007) [reply src/pymedphys/cli/dicom.py]: changelog?

Comment (SimonBiggs) [reply src/pymedphys/cli/dicom.py]: Cheers :) I'll put this back to WIP and I have been wanting to add in a suite of documentation + changelogs for the logfile side of things. Especially considering I plan to make the trf decode side of it public soon.

I'll land a heap of docs in this branch. Now's a better time than any other :)

Comment (Centrus007) [reply src/pymedphys/cli/dicom.py]: Yeah. I've actually been thinking that we ought to really spend some time updating and prettifying the docs.

I'm now starting the think docs are essential - almost as important as the code itself. Physicists will rely very heavily on the documentation when adopting PyMedPhys. This is something that James Kerns has done *very* well with `pylinac`.

**Comment by SimonBiggs** [comment]
@Centrus007, these docs are only a splattering of what needs to be there. But this pull request is getting quite large. I'd be keen to merge this and continue work on the docs on another branch.

Will likely pour quite a bit of time into docs in that other branch tomorrow and the day after.

Cheers,
Simon

**Comment by SimonBiggs** [comment]
@Centrus007 ready for review.

File: changelog.md [hunk 2]
```diff
@@ -32,36 +36,47 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
     `DicomCollection.from_directory(directory_path)` which would pull all DICOM
     files nested within a directory and sort them into `DicomStudy` objects
     based on their header UIDs.
+- Added CLI commands for a WIP docker server, logfile orchetration, and DICOM
+  editor tools.
+- Added a range of xlwings tools that allow the use of PyMedPhys functions
+  within Excel
+- Added the rudementry code enabling the pulling of profiles from Mephysto
```
Comment (Centrus007) [inline changelog.md]:
```suggestion
- Added rudimentary code to pull profiles from Mephysto
```

File: changelog.md [hunk 3]
```diff
@@ -32,36 +36,47 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
     `DicomCollection.from_directory(directory_path)` which would pull all DICOM
     files nested within a directory and sort them into `DicomStudy` objects
     based on their header UIDs.
+- Added CLI commands for a WIP docker server, logfile orchetration, and DICOM
```
Comment (Centrus007) [inline changelog.md]:
```suggestion
- Added CLI commands for a WIP docker server, logfile orchestration, and DICOM
```

File: changelog.md (hunk 2 above)
Comment (SimonBiggs) [reply changelog.md]: Thanks Matt, I think I should find a VSCode extension for spell check...

**Comment by SimonBiggs** [comment]
I'll let you merge this one, so that it attaches your user to the merge commit.

**Comment by SimonBiggs** [comment]
Gee the difference here hurts:

image

image

**Pull request merged**